std/haxe/format/JsonParser.hx

```diff
@@ -222,7 +236,7 @@ class JsonParser {
 					if (minus) minus = false;
 					digit = true; zero = false;
 				case '.'.code :
-					if (minus || point) invalidNumber(start);
+					if (minus || point || e) invalidNumber(start);
```

Comment:
what's this `e`? could you please add a test for this too?

Reply:
According to the json spec, the exponential is an integer. So you can't have something like `1e2.5`. I'll try to add a test for that tomorrow.